packages/firestore/test/unit/specs/spec_test_runner.ts

```diff
@@ -1279,6 +1281,12 @@ export async function runSpec(
   let count = 0;
   try {
     await sequence(steps, async step => {
+      assert(
+        step.clientIndex === undefined || tags.indexOf(MULTI_CLIENT_TAG) !== -1,
+        "Cannot use 'client()' to initialize a test that is not tagged with " +
+          "'multi-client'. Did you mean to use 'spec()'?"
+      );
```

Comment:
I wonder if it'd make sense to assert the inverse as well (if this is a multi-client test, then clientIndex must be defined)?  Basically:
```
const isMultiClientTest = tags.indexOf(MULTI_CLIENT_TAG) !== -1;
const hasClientIndex = step.clientIndex !== undefined;
assert(isMultiClientTest === hasClientIndex, "Tests can use client() if-and-only-if they are tagged with 'multi-client'.");
```

I don't care strongly if that seems pointless though.

Reply:
It's not pointless, but not necessarily worth opening an IDE for and fixing since it would not break anything. I'll leave it as an imaginary TODO :)